Source/JavaScriptCore/runtime/JSBigInt.cpp

```diff
@@ -1439,15 +1490,68 @@ JSBigInt::ComparisonResult JSBigInt::compare(JSBigInt* x, JSBigInt* y)
 {
     return compareImpl(HeapBigIntImpl { x }, HeapBigIntImpl { y });
 }
+
 JSBigInt::ComparisonResult JSBigInt::compare(int32_t x, JSBigInt* y)
 {
     return compareImpl(Int32BigIntImpl { x }, HeapBigIntImpl { y });
 }
+
 JSBigInt::ComparisonResult JSBigInt::compare(JSBigInt* x, int32_t y)
 {
     return compareImpl(HeapBigIntImpl { x }, Int32BigIntImpl { y });
 }
 
+JSBigInt::ComparisonResult JSBigInt::compare(JSBigInt* x, int64_t y)
+{
+    return compareImpl(HeapBigIntImpl { x }, Int64BigIntImpl { y });
+}
+
+JSBigInt::ComparisonResult JSBigInt::compare(JSValue x, int64_t y)
+{
+    ASSERT(x.isBigInt());
+#if USE(BIGINT32)
+    if (x.isBigInt32())
+        return compareImpl(Int32BigIntImpl { x.bigInt32AsInt32() }, Int64BigIntImpl { y });
+#endif
+    return compare(x.asHeapBigInt(), y);
+}
+
+JSBigInt::ComparisonResult JSBigInt::compare(JSBigInt* x, uint64_t y)
+{
+    return compareImpl(HeapBigIntImpl { x }, Int64BigIntImpl { y });
+}
+
+JSBigInt::ComparisonResult JSBigInt::compare(JSValue x, uint64_t y)
+{
+    ASSERT(x.isBigInt());
+#if USE(BIGINT32)
+    if (x.isBigInt32())
+        return compareImpl(Int32BigIntImpl { x.bigInt32AsInt32() }, Int64BigIntImpl { y });
+#endif
+    return compare(x.asHeapBigInt(), y);
+}
+
+JSBigInt::ComparisonResult JSBigInt::compare(JSValue x, JSValue y)
+{
+    ASSERT(x.isBigInt() && y.isBigInt());
+#if USE(BIGINT32)
+    if (x.isBigInt32() && y.isBigInt32()) {
+        int32_t x1 = x.asBigInt32();
+        int32_t y1 = y.asBigInt32();
```

Comment:
Can you make sure your other patch has tests for this bug?

Reply:
`USE_BIGINT32` is not enabled.

Reply:
Oh, whoops! Never mind me, I thought it was.